services/apps/portal/src/application/application-document/application-document.service.ts

```diff
@@ -0,0 +1,91 @@
+import { MultipartFile } from '@fastify/multipart';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Any, Repository } from 'typeorm';
+import { DocumentService } from '../../alcs/document/document.service';
+import { User } from '../../user/user.entity';
+import { ApplicationService } from '../application.service';
+import {
+  ApplicationDocument,
+  DOCUMENT_TYPE,
+} from './application-document.entity';
+
+@Injectable()
+export class ApplicationDocumentService {
+  constructor(
+    private documentService: DocumentService,
+    private applicationService: ApplicationService,
+    @InjectRepository(ApplicationDocument)
+    private applicationDocumentRepository: Repository<ApplicationDocument>,
+  ) {}
+
+  async attachDocument(
+    fileNumber: string,
+    file: MultipartFile,
+    user: User,
+    documentType: DOCUMENT_TYPE,
+  ) {
+    const application = await this.applicationService.getOrFail(fileNumber);
```

Comment:
obviously out of scope of this ticket, but I wonder if in case of portal we need to wrap all PG related errors in something user friendly so we do not scare applicants